docs/block-processing/overall.md

```diff
@@ -0,0 +1,105 @@
+# Block processing
+
+## Task definition
+
+Block processing can be described by presenting an algorithm to solve the following problem:
+
+> Given a sequence of blocks `B₀, B₁, B₂, …` (where `B₀` is the first
+genesis block) check whether block payload from these blocks are valid.
+
+We describe block processing as stateful algorithm:
+* Initial state `S₀` is derived from blockchain genesis data (see [mainnet genesis JSON](https://raw.githubusercontent.com/input-output-hk/cardano-sl/e7cfb1724024e0db2f25ddd2eb8f8f17c0bc497f/node/mainnet-genesis.json))
+* `S₁, S₂, …` are maintained as sequential application of blocks  `B₀, B₁, B₂, … ` to state `S₀`
+* State transition function. Given GState `S` and a main block `B` return either an error describing why `B` is invalid (w.r.t. tx payload) or new GState `S'`
```

Comment:
"GState" is undefined here. Either define it or use "state".

Reply:
@gromakovsky It wasn't defined in initial `txp.md` :(

Reply:
It was defined in my initial version https://github.com/input-output-hk/cardano-sl/blob/25e8136aa58acf4f2a53b9aa6853bde20fe75604/docs/txp.md#global-transaction-processing